### ISSUE TITLE
Structured learning test testStructuredLearningTrackingHytraFluo-N2DH-SIM-01.py fails occasionally

### ISSUE THREAD

**Issue by k-dominik** [ilastik/ilastik#1535]
I have observed that ```testStructuredLearningTrackingHytraFluo-N2DH-SIM-01.py``` fails occasionally using cplex or gurobi on linux.

I have run this test-file 20 times with both, cplex and gurobi and found that there was a failure in 7 times with cplex and 4 times with gurobi, respectively. (I have confirmed this behaviour with gurobi on a second machine, resulting in 5 out of 20).

The test that is failing is ```TestStructuredLearningTrackingHeadless.testCSVExport``` with the following message:

```pytb
Traceback (most recent call last):
  File ".../miniconda2/envs/cplex-test/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File ".../miniconda2/envs/cplex-test/ilastik-meta/lazyflow/lazyflow/utility/timer.py", line 142, in wrapper
    return func(*args, **kwargs)
  File ".../ilastik-meta-repo/ilastik/tests/test_applets/structuredLearningTracking/testStructuredLearningTrackingHytraFluo-N2DH-SIM-01.py", line 208, in testCSVExport
    assert false_detection_count == self.EXPECTED_FALSE_DETECTIONS_NUM, 'Number of false detections {} in the csv file differs from expected {}.'.format(false_detection_count,self.EXPECTED_FALSE_DETECTIONS_NUM)
AssertionError: Number of false detections 3 in the csv file differs from expected 1.
```

I could observe the following numbers of false detections: {2, 3, 4, 19, 21, 36}.

**Comment by ilastikdev** [comment]
fixed in 4dd56947bb25707c53947cc838488c31057bc7d3

see https://github.com/ilastik/ilastik/pull/1537

**Comment by ilastikdev** [comment]
from dominik:

"
Hi Janez,

I have merged your most recent commit with the modified ilp file and ran the test 40 
times without any error. I think we can assume now it works again.

Cool, that you fixed it!

Cheers
Dominik
"
